[WebGPU] GPUTexture.createView should report useful error messages when validation fails

mwyrzykowski opened the pull request:
#### 3cd7a4b56a4a060a1a7cfac76c43b38af5d6abf7
<pre>
[WebGPU] GPUTexture.createView should report useful error messages when validation fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=265930">https://bugs.webkit.org/show_bug.cgi?id=265930</a>
&lt;radar://119235084&gt;

Reviewed by Tadeu Zagallo.

http/tests/webgpu/webgpu/api/validation/createView.html now passes locally
after not reporting validation errors on destroyed textures.

Also report some more useful error messages to enable better diagnosing of failures
when they occur.

* LayoutTests/http/tests/webgpu/webgpu/api/validation/createView-expected.txt:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::errorValidatingTextureViewCreation const):
(WebGPU::Texture::createView):
(WebGPU::Texture::destroy):
(WebGPU::Texture::validateCreateView const): Deleted.

Canonical link: <a href="https://commits.webkit.org/271627@main">https://commits.webkit.org/271627@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4578dbe909b39d5fbc75e958eff336efb27cd9a1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29055 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7727 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30405 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31666 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9856 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5040 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26476 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29327 "Passed tests") | [💥 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6387 "An unexpected error occured. Recent messages:OS: Sonoma (14.0), Xcode: 15.0; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/24923 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5542 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/5669 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/25950 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33003 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26547 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/26388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/31917 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/5646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/3823 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/29700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7306 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6930 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6147 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/6167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WebGPU/Texture.mm:
```diff
@@ -2188,92 +2188,92 @@ static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonP
     if (m_sampleCount > 1) {
         if (descriptor.dimension != WGPUTextureViewDimension_2D)
-            return false;
+            return @"createView: m_sampleCount > 1 && descriptor.dimension != WGPUTextureViewDimension_2D";
```
Just a nit, but are some of these errors maybe a bit too low level? Or it wouldn't make sense to make it more descriptive instead of code?

mwyrzykowski replied:
This is a good point, it should probably be written like:
```
@"GPUTexture.createView: sampleCount > 1 and dimension != 2D";
```

mwyrzykowski replied:
I will clean that up

mwyrzykowski committed:
[WebGPU] GPUTexture.createView should report useful error messages when validation fails
https://bugs.webkit.org/show_bug.cgi?id=265930
<radar://119235084>

Reviewed by Tadeu Zagallo.

http/tests/webgpu/webgpu/api/validation/createView.html now passes locally
after not reporting validation errors on destroyed textures.

Also report some more useful error messages to enable better diagnosing of failures
when they occur.

* LayoutTests/http/tests/webgpu/webgpu/api/validation/createView-expected.txt:
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::errorValidatingTextureViewCreation const):
(WebGPU::Texture::createView):
(WebGPU::Texture::destroy):
(WebGPU::Texture::validateCreateView const): Deleted.

Canonical link: https://commits.webkit.org/271627@main
